test/object-store/c_api/c_api.cpp

```diff
@@ -3962,6 +3962,163 @@ struct Userdata {
 };
 
 #if REALM_ENABLE_SYNC
+
+TEST_CASE("C API - client reset", "[c_api][client-reset]") {
+    reset_utils::Partition partition{"realm_id", random_string(20)};
+    Property partition_prop = {partition.property_name, PropertyType::String | PropertyType::Nullable};
+    Schema schema{
+        {"object",
+         {
+             {"_id", PropertyType::ObjectId, Property::IsPrimary{true}},
+             {"value", PropertyType::Int},
+             partition_prop,
+         }},
+    };
+
+    std::string base_url = get_base_url();
+    REQUIRE(!base_url.empty());
+    auto server_app_config = minimal_app_config(base_url, "c_api_client_reset_tests", schema);
+    server_app_config.partition_key = partition_prop;
+    TestAppSession test_app_session(create_app(server_app_config));
+
+    auto app = test_app_session.app();
+    auto get_valid_config = [&]() -> SyncTestFile {
+        create_user_and_log_in(app);
+        return SyncTestFile(app->current_user(), partition.value, schema);
+    };
+    SyncTestFile local_config = get_valid_config();
+    SyncTestFile remote_config = get_valid_config();
+
+    struct ClientSyncObject {
+        static ClientSyncObject& instance()
+        {
+            static ClientSyncObject client;
+            return client;
+        }
+        bool stop()
+        {
+            return m_stop.load();
+        }
+        void signal()
+        {
+            m_stop = true;
+        }
+        std::atomic_bool m_stop{false};
+    };
+
+    auto make_reset = [&](Realm::Config config_local,
+                          Realm::Config config_remote) -> std::unique_ptr<reset_utils::TestClientReset> {
+        return reset_utils::make_baas_client_reset(config_local, config_remote, test_app_session);
+    };
+
+    realm_sync_config_t* local_sync_config = static_cast<realm_sync_config_t*>(local_config.sync_config.get());
+
+    SECTION("Manual reset") {
+        realm_sync_config_set_resync_mode(local_sync_config, RLM_SYNC_SESSION_RESYNC_MODE_MANUAL);
+
+        struct ResetRealmFiles {
+            void set_app(std::shared_ptr<realm::app::App> app)
+            {
+                m_app = app;
+            }
+            void do_reset(const char* path)
+            {
+                realm_app_t realm_app{m_app};
+                realm_sync_immediately_run_file_actions(&realm_app, path);
+            }
+            static ResetRealmFiles& instance()
+            {
+                static ResetRealmFiles instance;
```

Comment:
Why is a static needed here?

Reply:
Same as above.
A possible alternative is to have a global function with a global variable.